streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -125,6 +127,8 @@ public void commit() {
                 // and the state current offset would be used to checkpoint
                 stateMgr.checkpoint(Collections.emptyMap());
 
+                changelogOffsetSnapshot = stateMgr.changelogOffsets();
```

Comment:
Should we make a copy of the map, to guard against the state manager implementation?

Reply:
Yup, that's a better idea.